projects/components/select/src/testing/select.harness.ts

```diff
@@ -0,0 +1,208 @@
+    return (await this._matSelect()).focus();
+  }
+
+  /** Blurs the select and returns a void promise that indicates when the action is complete. */
+  async blur(): Promise<void> {
+    return (await this._matSelect()).blur();
+  }
+
+  /** Whether the select is focused. */
+  async isFocused(): Promise<boolean> {
+    return (await this._matSelect()).isFocused();
+  }
+
+  /** Gets the options inside the select panel. */
+  async getOptions(filter: Omit<OptionHarnessFilters, 'ancestor'> = {}): Promise<MatOptionHarness[]> {
+    const options = await (await this._matSelect()).getOptions(filter);
+    const result = [];
+    for (const option of options) {
+      const host = await option.host();
+      if (await host.hasClass('ps-select__search')) {
+        continue;
+      }
+      result.push(option);
+    }
+    return result;
+  }
+
+  // OptionGroups are not supported at the moment
+  //   /** Gets the groups of options inside the panel. */
+  //   async getOptionGroups(filter: Omit<OptgroupHarnessFilters, 'ancestor'> = {}): Promise<MatOptgroupHarness[]> {
+  //     return (await this._matSelect()).getOptionGroups(filter);
+  //   }
+
+  /** Gets whether the select is open. */
+  async isOpen(): Promise<boolean> {
+    return (await this._matSelect()).isOpen();
+  }
+
+  /** Opens the select's panel. */
+  async open(): Promise<void> {
+    return (await this._matSelect()).open();
+  }
+
+  /**
+   * Clicks the options that match the passed-in filter. If the select is in multi-selection
+   * mode all options will be clicked, otherwise the harness will pick the first matching option.
+   */
+  async clickOptions(filter: OptionHarnessFilters = {}): Promise<void> {
+    await this.open();
+
+    const [isMultiple, options] = await Promise.all([this.isMultiple(), this.getOptions(filter)]);
+
+    if (options.length === 0) {
+      throw Error('Select does not have options matching the specified filter');
+    }
+
+    if (isMultiple) {
+      await Promise.all(options.map((option) => option.click()));
+    } else {
+      await options[0].click();
+    }
+  }
+
+  /** Closes the select's panel. */
+  async close(): Promise<void> {
+    return (await this._matSelect()).close();
+  }
+
+  /** Gets a boolean promise indicating if the select is in the error state. */
+  async isErrorState(): Promise<boolean> {
+    return await (await this.host()).hasClass('ps-select-invalid');
+  }
+
+  /** Gets the option that clears the selected value. */
+  async getEmptyOption() {
+    const options = await this.getOptions({ text: '--' });
+    for (const option of options) {
+      if (await (await option.host()).hasClass('ps-select__empty-option')) {
+        return option;
+      }
+    }
+
+    return null;
+  }
+
+  /** Gets the option that shows the error message. */
+  async getErrorOption() {
+    const options = await this.getOptions();
+    for (const option of options) {
+      if (await (await option.host()).hasClass('ps-select__error')) {
```

Comment:
I couldn't find any definition for this class? Is this just a marker class?

Reply:
yes, same for ps-select__search